app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java

```diff
@@ -1,15 +1,359 @@
+    @BindView(R.id.ctrl_btn_wave1)
+    Button btnCtrlWave1;
+    @BindView(R.id.ctrl_btn_wave2)
+    Button btnCtrlWave2;
+    @BindView(R.id.ctrl_btn_freq)
+    Button btnCtrlFreq;
+    @BindView(R.id.ctrl_btn_phase)
+    Button btnCtrlPhase;
+    @BindView(R.id.ctrl_img_btn_sin)
+    ImageButton imgBtnSin;
+    @BindView(R.id.ctrl_img_btn_sq)
+    ImageButton imgBtnSq;
+    @BindView(R.id.ctrl_img_btn_tri)
+    ImageButton imgBtnTri;
+
+    //seek bar controls
+    @BindView(R.id.img_btn_up)
+    ImageButton imgBtnUp;
+    @BindView(R.id.img_btn_down)
+    ImageButton imgBtnDown;
+    @BindView(R.id.seek_bar_wave_gen)
+    IndicatorSeekBar seekBar;
+    @BindView(R.id.btn_set)
+    Button btnSet;
+
+    private int leastCount,seekMax, seekMin;
+    private String unit;
+
+    //const values
+    public static final int SIN = 1;
+    public static final int TRIANGULAR = 2;
+    public static final int SQUARE = 3;
+    public static final String WAVETYPE = "wavetype";
+    public static final String WAVE1 = "wave1";
+    public static final String WAVE2 = "wave2";
+    public static final String FREQUENCY = "frequency";
+    public static final String PHASE = "phase";
+    public static final String DUTY = "duty";
+
+    private static String btn_Active = "";
+    private static String prop_active = "";
+
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_wave_generator);
+        ButterKnife.bind(this);
+        new WaveGeneratorCommon();
+
+        selectWave1();    //on starting wave1 will be selected
+
+        btnCtrlWave1.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (!btn_Active.equals(WAVE1)) {
+                    selectWave1();
+                }
+            }
+        });
+
+        btnCtrlWave2.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (!btn_Active.equals(WAVE2)) {
+                    selectWave2();
+                }
+            }
+        });
+
+        imgBtnSin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                WaveGeneratorCommon.wave.get(btn_Active).put(WAVETYPE, SIN);
+                selectWaveform(SIN);
+            }
+        });
+        imgBtnSq.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                WaveGeneratorCommon.wave.get(btn_Active).put(WAVETYPE, SQUARE);
+                selectWaveform(SQUARE);
+            }
+        });
+        imgBtnTri.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                WaveGeneratorCommon.wave.get(btn_Active).put(WAVETYPE, SQUARE);
```

Comment:
Is this supposed to be Triangular instead of Square?

Reply:
I will make that square but changing the phase for wave1 was not there in first place in control module